src/main/java/bio/terra/cli/command/resources/create/GcsBucket.java

```diff
@@ -0,0 +1,83 @@
+package bio.terra.cli.command.resources.create;
+
+import bio.terra.cli.command.helperclasses.BaseCommand;
+import bio.terra.cli.command.helperclasses.FormatOption;
+import bio.terra.cli.service.WorkspaceManager;
+import bio.terra.workspace.model.AccessScope;
+import bio.terra.workspace.model.CloningInstructionsEnum;
+import bio.terra.workspace.model.GcpGcsBucketDefaultStorageClass;
+import bio.terra.workspace.model.ResourceDescription;
+import java.util.Collections;
+import picocli.CommandLine;
+
+/** This class corresponds to the fourth-level "terra resources create gcs-bucket" command. */
+@CommandLine.Command(
+    name = "gcs-bucket",
+    description = "Add a controlled GCS bucket.",
+    showDefaultValues = true)
+public class GcsBucket extends BaseCommand {
+  @CommandLine.Option(
+      names = "--name",
+      required = true,
+      description =
+          "Name of the resource, scoped to the workspace. Only alphanumeric and underscore characters are permitted.")
+  private String name;
+
+  @CommandLine.Option(names = "--description", description = "Description of the resource")
+  private String description;
+
+  @CommandLine.Option(
+      names = "--cloning",
+      description =
+          "Instructions for handling when cloning the workspace: ${COMPLETION-CANDIDATES}")
+  private CloningInstructionsEnum cloning = CloningInstructionsEnum.NOTHING;
+
+  @CommandLine.Option(
+      names = "--access",
+      description = "Access scope for the resource: ${COMPLETION-CANDIDATES}")
+  private AccessScope access = AccessScope.SHARED_ACCESS;
```

Comment:
Consider using MixIns for each of these "standard" fields across different resources. That would let us define them once and reuse them for each resource.

It might even make sense to use a ResourceName mixin in not only create commands, but also delete & update, anywhere we use a "resource name." Been thinking about using more mixins for some of my shared notebook options, e.g. `--location`, next time I'm touching them.

Reply:
Great suggestion! Added three new mixin classes: `ResourceName`, `CreateResource`, `CreateControlledResource` that contain varying combinations of options used in multiple commands.